src/processing/strategies/transform.rs

```diff
@@ -0,0 +1,37 @@
+use crate::processing::strategies::{
+    CommitRequest, InvalidMessage, MessageRejected, ProcessingStrategy,
+};
+use crate::types::Message;
+
+pub struct Transform<TPayload: Clone + Send + Sync, TTransformed: Clone + Send + Sync> {
+    pub function: fn(Message<TPayload>) -> Result<TTransformed, InvalidMessage>,
+    pub next_step: Box<dyn ProcessingStrategy<TTransformed>>,
+}
+
+impl<TPayload: Clone + Send + Sync, TTransformed: Clone + Send + Sync> ProcessingStrategy<TPayload>
+    for Transform<TPayload, TTransformed>
+{
+    fn poll(&mut self) -> Option<CommitRequest> {
+        None
```

Comment:
Shouldn't we poll on the next step? Otherwise nobody is doing that at any time.

Reply:
Yes you are right